homeassistant/components/vicare/button.py

```diff
@@ -72,20 +72,20 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Create the ViCare button entities."""
-    api = hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
-
     entities = []
 
-    for description in BUTTON_DESCRIPTIONS:
-        entity = await hass.async_add_executor_job(
-            _build_entity,
-            description.name,
-            api,
-            hass.data[DOMAIN][config_entry.entry_id][VICARE_DEVICE_CONFIG],
-            description,
-        )
-        if entity is not None:
-            entities.append(entity)
+    for device_config in hass.data[DOMAIN][config_entry.entry_id][DEVICE_CONFIG_LIST]:
+        device = get_device(config_entry, device_config)
```

Comment:
Instead of creating the device in each platform, we could store it in `hass.data[DOMAIN][config_entry.entry_id]`